app/controllers/admin/patient_imports_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Admin::PatientImportsController < AdminController
+  include FileUploadable
+
+  def new
+    authorize { current_admin.power_user? }
+  end
+
+  def create
+    authorize { current_admin.power_user? }
+
+    facility = Facility.find(params[:facility_id])
+
+    data = read_xlsx_or_csv_file(params[:patient_import_file])
+    params = PatientImport::SpreadsheetTransformer.call(data, facility: facility)
+    errors = PatientImport::Validator.new(params).errors
+
+    if errors.values.flatten.any?
+      @errors = errors
+      render :new
+    else
+      results = PatientImport::Importer.call(params: params, facility: facility, admin: current_admin)
+      redirect_to new_admin_patient_import_url, notice: "Successfully imported #{results.count} patients to #{facility.name}"
```

Comment:
It may be good to also display the number of total rows from the spreadsheet, especially so the user running things will see how many rows got skipped / ignored due to any errors.

Reply:
Great idea. I will make the `results` more useful. We should know:
* How many total patients I just tried to import
* How many were imported
* Which rows were not imported.

This will also be easier to craft when it's an email via background job 😉

Reply:
This point is now moot, since every patient gets their own job, and we'll be inspecting success of the import directly in Sidekiq.